doc/services/min_heap/index.rst

```diff
@@ -0,0 +1,68 @@
+.. _min_heap:
+
+Min-Heap Data Structure
+#######################
+
+.. contents::
+  :local:
+  :depth: 2
+
+The Min-Heap implementation provides an efficient data structure for
+managing a dynamically changing list of elements while maintaining the ability
+to quickly extract the minimum value. It's a tree-based data structure that
+satisfies the heap property and supports common operations such as insertion,
+removal and popping the minimum element from the Min-Heap
+
+This section explains the motivation behind the implementation, its internal
+structure, API usage, and example scenarios for embedded systems and real-time
+environments.
+
+Heap Structure
+**************
+
+The heap is maintained as a complete binary tree stored in an array.
+Each node satisfies the **min-heap** property:
+
+   - The value of each node is less than or equal to the values of its children.
+
+This property ensures that the **minimum element is always at the root**
+(index 0).
+
+.. code-block:: text
+
+    Index:      0   1   2   3   4   5   6
+    Values:    [1,  3,  5,  8,  9, 10, 12]
+
+                   1
+                 /   \
+               3       5
+              / \     / \
+             8   9  10  12
+
+For any node at index ``i``, its children are at indices:
+- Left child:  ``2*i + 1``
+- Right child: ``2*i + 2``
+
+Its parent is at index:
+- Parent: ``(i - 1) / 2``
```

Comment:
also note that if you want you can use nicely formatted math formulas (https://docs.zephyrproject.org/latest/contribute/documentation/guidelines.html#mathematical-formulas)

```suggestion
Its parent is at index:

- Parent: ``(i - 1) / 2``
```